src/ray/core_worker/core_worker.cc

```diff
@@ -4327,6 +4327,22 @@ void CoreWorker::RecordTaskLogEnd(int64_t stdout_end_offset,
       worker::TaskStatusEvent::TaskStateUpdate(task_log_info));
 }
 
+void CoreWorker::UpdateTaskIsDebuggerPaused(const TaskID &task_id,
+                                            const bool is_debugger_paused) {
+  absl::MutexLock lock(&mutex_);
+  auto current_task_it = current_tasks_.find(task_id);
+  RAY_CHECK(current_task_it != current_tasks_.end())
+      << "We should have set the current task spec before executing the task.";
+  RAY_LOG(DEBUG) << "Task " << current_task_it->second.TaskId()
+                 << " is paused by debugger set to" << is_debugger_paused;
+  task_manager_->RecordTaskStatusEvent(
+      current_task_it->second.AttemptNumber(),
+      current_task_it->second,
+      rpc::TaskStatus::RUNNING,
```

Comment:
let's do `rpc::TaskStatus::NIL` since we are not updating task status for debugger?

Reply:
I don't think I fully understand how `RecordTaskStatusEvent` works so I will leave that up to you. Is replacing it with nil will ignore updating the state or reset it?

Reply:
Alright did some testing, replacing with nil seems harmless. I updated it.